### PR TITLE
Projects category page redesign

### PR DESCRIPTION
**Pull request by adrianthedev**
It's not finished yet. I've hit some snags on how the filtering system should work. It was a bit tricky to figure out how to jump from categories to our regular filtering system.

I created a sort of alpine component that handles the filters and creates the right URL for the filters to work.

How I thought about it was. We have a category page `/medical` where we fill in the `project_types` filter in the background with `@applied_filters` (variable name TBD). Then, if the user changes one of the other filters (skills, countries, status, etc) we keep him on that path with the new filters applied to the URL `/medical?status=SOME_STATUS&countries[]=SOME_COUNTRY`.
But if the user changes the `project_types` filter, then we apply all of the filters on the `/projects` path as usual (`/medical?status=SOME_STATUS&countries[]=SOME_COUNTRY`) because then we don't fall in one particular category.
One braking change in this PR, is how we add the array filters to the URL. In the past we did one key and split the value by commas (`?project_skills=SKILL_ONE,SKILL_TWO`) and now we add multiple keys with `[]` and each value to it (`?project_skills[]=SKILL_ONE&skills[]=SKILL_TWO`). This is how the URL parses package works, but I've read about it and that's how it kinda should be.

Left to do:
 - [x] make the back-end respond to the new filters
 - [x] figure out a way of displaying the featured posts on "All categories"
 - [x] mobile version of the filters and categories nav
 - [x] misc design tweaks to bring it closer to the design
 - [x] show the category `project_types` under the description
 - [x] make sure that the previous filtering links still work (`project_types=Scale+testing%2CMedical+facilities%2CMedical+equipments%2CTreatment+R%26D` gets processed as `project_types[]=Treatment%20R%26D&project_types[]=Medical%20equipments&project_types[]=Medical%20facilities`)
 - [x] add the new "All Categories" icon from @tinnei. I left a comment on figma with what we need.
 - [x] make sure that the rest of the pages that use common components (pagination, header announcement, etc.) still look good. I might have borked some of them in the process.
 - [ ] ~add icons to the filters bar labels (Search, Sort by, Filter by)~

Fixes https://github.com/helpwithcovid/covid-volunteers/issues/100

### CODE REVIEW

File: app/views/projects/_category-info.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<%
+  if @project_category.blank?
```
Comment (sradu) [inline app/views/projects/_category-info.html.erb:2]: should probably be a helper method

File: app/views/projects/_filter-bar-filter.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    </div>
+    <div class="bg-white">
+      <button class="button-indigo w-full mt-2" @click="applyFilters()">Apply filters</button>
+      <button class="w-full mt-2" @click="resetFilters()">Reset filters</button>
+    </div>
+  </div>
+</div>
+
+<script>
+  function dropdown<%= filter.camelize %>() {
```
Comment (sradu) [inline app/views/projects/_filter-bar-filter.html.erb:22]: This doesn't feel right anymore, js in HTML.

Comment (adrianthedev) [reply app/views/projects/_filter-bar-filter.html.erb:22]: Yeah, I know what you mean. We can extract that for sure. It's just a first draft of the code.

**Comment by jamiew** [comment]
Sorry about that, a shell script of mine accidentally deleted remote branch instead of local

**Comment by tinnei** [comment]
@adrianthedev consolidated feedback from Cathy and I in the doc here:
https://docs.google.com/document/d/12v9219yJkDqGdHjEmHT7rqpsNGFZiPQBCRpA84-Wjl8/edit?usp=sharing

The most pressing update needs are listed in the first table. And small notes/good to have's are in the second table. Tag us in a comment if you need any more clarification! 

Also definitely open to trim the tasks into other PR and ask for help!

**Pull request merged**